packages/venia-concept/src/shared/propShapes.js

```diff
@@ -0,0 +1,19 @@
+import { shape, number, string, bool } from 'prop-types';
```

Comment:
I'm very reluctant to add a "shared proptypes" file. We did this at my previous employer while using GraphQL. We ended up with dozens of permutations of similar-but-slightly-different shapes for each component. We also found it encouraged over-fetching of data since you didn't have to make a 1-off version of a shared proptype.

The other downside was that you no longer can just look at the file and see the shape of data. You now need to jump through the abstraction to the shared prop files, and hop back and forth between the component and the proptype implementation.

Reply:
We have to find a compromise on introducing high-level abstraction. They're not always necessary, no, but it's necessary to create some. We have to lower cognitive load for developers, and busy, deep inline definitions are cognitive load.

I see how shared propType shapes could become annoying if you define shared shapes too early, but we will have a shared prop shapes file. *It's the equivalent of GraphQL fragments.* (In fact, we might later put them together in a convenient wrapper, but not yet.) We can debate what belongs there, but we will need higher-level objects. This exports actual immutable prop types, not plain objects for `PropTypes.shape()`. If a dev needs a different shape, they have to make it inline. Whereas if they need the same shape, they use the shared shape.

It may make things a little harder for maintainers and code reviewers in exchange, but in cases where that's a tradeoff, our project's mission is to lean in the third party dev's direction. We should keep this file.

Reply:
I have to agree with @DrewML. This is a path to the dark side. If the product is ever ripe for this optimization, that'll be clear in the future and it'll make a good PR then.

As you said, if a developer needs a different shape, they have to make it inline. That's going to encourage them to use the wrong shape for convenience. I believe we want people to write granular, specific, custom queries on a per-component basis, not use the most convenient existing query.